--- OpenTTD_base/src/rev.cpp.in
+++ OpenTTD_work/src/rev.cpp.in
@@ -38,6 +38,11 @@
 const std::string _openttd_revision = "${REV_VERSION}";
 
 /**
+ * The text version of NodusTTD's patchset revision.
+ */
+const std::string _nodusttd_revision = "${REV_NODUSTTD_VERSION}";
+
+/**
  * The text version of OpenTTD's build date.
  * Updating the build date in this file is the safest as it generally gets
  * updated for each revision in contrary to most other files that only see
