--- OpenTTD_base/src/settingentry_gui.cpp
+++ OpenTTD_work/src/settingentry_gui.cpp
@@ -670,6 +670,7 @@
 				general->Add(new SettingEntry("gui.window_soft_limit"));
 				general->Add(new SettingEntry("gui.right_click_wnd_close"));
 				general->Add(new SettingEntry("gui.toolbar_dropdown_autoselect"));
+				general->Add(new SettingEntry("gui.cursor_rendering_mode"));
 			}
 
 			SettingsPage *viewports = interface->Add(new SettingsPage(STR_CONFIG_SETTING_INTERFACE_VIEWPORTS));
