--- OpenTTD_base/os/emscripten/pre.js
+++ OpenTTD_work/os/emscripten/pre.js
@@ -85,6 +85,10 @@
     }
 
     var leftButtonDown = false;
+    document.addEventListener("DOMContentLoaded", e => {
+        if ('serviceWorker' in navigator) navigator.serviceWorker.register('/sw.js');
+        Module.onWelcome();
+    });
     document.addEventListener("mousedown", e => {
         if (e.button == 0) {
             leftButtonDown = true;
